### PR TITLE
Adds SwWebhookAuthentication middleware

### PR DESCRIPTION
**Pull request by jpsantosbh**
ref: (#3100)[https://github.com/signalwire/cloud-support/issues/3100]

check for a [reference usage](https://github.com/signalwire/signalwire-ruby-hmac-examples/pull/3)

Adds a SwWebhookAuthentication

### CODE REVIEW

File: lib/rack/signalwire_webhook_authentication.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'rack/media_type'
+
+module Rack
+
+  class SwWebhookAuthentication
```
Comment (ryanwi) [inline lib/rack/signalwire_webhook_authentication.rb]: can spell this out to match the filename.

Comment (jpsantosbh) [reply lib/rack/signalwire_webhook_authentication.rb]: Needs to be the other way around `SignalwireWebhookAuthentication` is too big... That's why I renamed but forgot the file name.

Comment (ryanwi) [reply lib/rack/signalwire_webhook_authentication.rb]: too big style-wise or function-wise?

Comment (jpsantosbh) [reply lib/rack/signalwire_webhook_authentication.rb]: I was getting an error running the app saying the the name was too big

Comment (ryanwi) [reply lib/rack/signalwire_webhook_authentication.rb]: odd, have never heard of any limits

Comment (jpsantosbh) [reply lib/rack/signalwire_webhook_authentication.rb]: Yep, weird I tried again with `SignalwireWebhookAuthentication` and both worked the spec and the demos

File: lib/rack/signalwire_webhook_authentication.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'rack/media_type'
+
+module Rack
+
+  class SignalwireWebhookAuthentication
+
```
Comment (ryanwi) [inline lib/rack/signalwire_webhook_authentication.rb]:
```suggestion
```

File: lib/rack/signalwire_webhook_authentication.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'rack/media_type'
+
+module Rack
+
```
Comment (ryanwi) [inline lib/rack/signalwire_webhook_authentication.rb]:
```suggestion
```

File: spec/rack/signalwire_webhook_authentication_spec.rb [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+      request['HTTP_X_SIGNALWIRE_SIGNATURE'] = 'TR9Skm9jiF4WVRJznU5glK5I83k='
+      expect(request['CONTENT_TYPE']).to eq('application/x-www-form-urlencoded')
+
+      status, headers, body = middleware.call(request)
+
+      expect(status).to be(200)
+    end
+
+  end
+end
```
Comment (ryanwi) [inline spec/rack/signalwire_webhook_authentication_spec.rb]:
```suggestion
end

```

**Comment by ryanwi** [comment]
Should this be a minor version bump you think? Update the version.rb file and add a Changelog entry?

**Review by ryanwi — APPROVED** [review]

**Pull request merged**